eZ/Publish/Core/Persistence/Cache/ContentHandler.php

```diff
@@ -131,6 +136,7 @@ public function setStatus($contentId, $status, $version)
         $return = $this->persistenceHandler->contentHandler()->setStatus($contentId, $status, $version);
 
         $this->cache->clear('content', $contentId, $version);
+        $this->cache->clear('content', 'info', $contentId, 'versioninfo', $version);
```

Comment:
can be moved to else condition below

Reply:
Are you sure? It makes sense if you clear the content in a certain version, that you would want to clear the version itself too. Version has a status property which reflects change done here.

Reply:
Maybe I'm reading it wrong, but seems on public all of them will be cleared by `$this->cache->clear('content', 'info', $contentId);`.

Reply:
Well, yes, but only because this method is called when publishing the content. But if it were called standalone, or in some other workflow, then there are cases where `if` wouldn't execute.

Reply:
And on top of that, it's actually used twice when publishing the content, once for archiving the currently published version, and once for publishing the current draft, so for that archived version info, you wouldn't get a  cache clear.

Reply:
I'm not following here I think.

What I'm saying is that becasue Stash is hiracical, this versions cache will be cleared by the info clear, so afaik what you have is idenitical to the following, but it will save a duplicated clear as far as I can see:
```
         if ($status === VersionInfo::STATUS_PUBLISHED) {
             $this->cache->clear('content', 'info', $contentId); // This will clear all version info as well, as it typically changes status of the one being archived to
             $this->cache->clear('content', 'info', 'remoteId');
         } else {
             $this->cache->clear('content', 'info', $contentId, 'versioninfo', $version);// Only one specific version info affected here, content info not affected (unless prevous status of the changed versionInfo was published pherhaps...)
         }
```

Reply:
Ah okay, so we misunderstood each other completely :) I was sure you were talking about moving the call to the `if` clause, altghough *now* I see that you wrote `else` :)

Updated now!

Reply:
Yeah, understood we where not talking about the same thing 👍 :)